From 4415539c2c0362c4cd4888276643251630fd856a Mon Sep 17 00:00:00 2001
From: Todd Poynor <toddpoynor@google.com>
Date: Wed, 18 Jul 2012 18:47:01 -0700
Subject: [PATCH 602/944] power: smb347: Stop charging on termination current
 status or irq

The IRQ bts often don't read as set for whatever reason.

Change-Id: Ic23c58da502280e7968410d677b06931a420aa68
Signed-off-by: Todd Poynor <toddpoynor@google.com>
---
 drivers/power/smb347-charger.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/power/smb347-charger.c b/drivers/power/smb347-charger.c
index d67d2c1..09cc839 100644
--- a/drivers/power/smb347-charger.c
+++ b/drivers/power/smb347-charger.c
@@ -813,15 +813,16 @@ static irqreturn_t smb347_interrupt(int irq, void *data)
 	 * If we reached the termination current the battery is charged.
 	 * Disable charging to ACK the interrupt and update status.
 	 */
-	if (irqstat_c & (IRQSTAT_C_TERMINATION_IRQ | IRQSTAT_C_TAPER_IRQ)) {
-		if (irqstat_c & IRQSTAT_C_TERMINATION_STAT) {
-			smb347_charging_disable(smb);
-			power_supply_changed(&smb->battery);
-		}
-
+	if (irqstat_c & (IRQSTAT_C_TERMINATION_IRQ |
+			 IRQSTAT_C_TERMINATION_STAT)) {
+		smb347_charging_disable(smb);
+		power_supply_changed(&smb->battery);
 		ret = IRQ_HANDLED;
 	}
 
+	if (irqstat_c & IRQSTAT_C_TAPER_IRQ)
+		ret = IRQ_HANDLED;
+
 	/*
 	 * If we got an under voltage interrupt it means that AC/USB input
 	 * was disconnected.
-- 
1.8.0

